lib/replace-in-file.js

```diff
@@ -60,33 +61,49 @@ module.exports = function replaceInFile(config, cb) {
   }
 
   //Initialize helper vars
-  var totalFiles = config.files.length;
+  var totalFiles = 0;
   var processedFiles = 0;
   var changedFiles = [];
 
   //No callback given? Perform sync operation
   if (!cb) {
     config.files.forEach(function(file) {
-      if (replaceSync(file, config.replace, config.with)) {
-        changedFiles.push(file);
-      }
+      //Expand globs
+      glob.sync(file, {nodir: true}).forEach(function(file) {
+        if (replaceSync(file, config.replace, config.with)) {
+          changedFiles.push(file);
+        }
+      });
     });
     return changedFiles;
   }
 
   //Otherwise, perform async operation
   config.files.forEach(function(file) {
-    replace(file, config.replace, config.with, function(error, hasChanged) {
-      if (error) {
-        return cb(error);
-      }
-      if (hasChanged) {
-        changedFiles.push(file);
+    //Expand globs
+    glob(file, {nodir: true}, function(err, files) {
+      if (err) {
+        return cb(err);
       }
-      processedFiles++;
-      if (processedFiles === totalFiles) {
-        cb(null, changedFiles);
+      //Glob doesn't error if no files match, so we do:
+      if (files.length === 0) {
+        return cb(new Error('No files match the pattern: ' + file));
```

Comment:
@adambuczynski True, the only issue is that since glob is processing all file paths, if none of the file paths exist, replace will fail silently (breaking the `return an error on failure` case [https://github.com/adambuczynski/replace-in-file/blob/master/test/replace-in-file.spec.js#L96]). Thoughts?